azure-proto-core/Shared/Autorest/CodeGenTypeAttribute.cs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+
+namespace Azure.Core
+{
+    [AttributeUsage(AttributeTargets.Class)]
```

Comment:
If this is copied from Azure.Core, it may not have the intended effect - the type identity for this type will be different, meaning any code that have that looks for this attribute will not find it.